// atm-card-react/src/lib/BackCardData.js
import * as React from 'react';
import { Rotate } from './icons';
import { cvvMask } from './utils';
import InputComponent from './InputComponent';
const BackCardData = (props) => {
    const card = React.useRef();
    const [cvvEdit, setCvvEdit] = React.useState(false);
    const cvvInput = React.useRef(null);
    const handleClick = (e) => {
        if (card.current.contains(e.target))
            return;
        setCvvEdit(false);
    };
    const handleKeyPress = (e) => {
        if (e.key === 'Enter')
            setCvvEdit(false);
    };
    React.useEffect(() => {
        if (props.cvv.length === 3)
            setCvvEdit(false);
        document.addEventListener('mousedown', handleClick, true);
        return () => {
            document.removeEventListener("mousedown", handleClick, true);
        };
    }, [props.cvv]);
    return (React.createElement("div", { className: "component-atm-card back", ref: card, style: {
            backgroundImage: `url(${props.bgImage ? props.bgImage : ''})`,
            backgroundColor: props.bgColor ? props.bgColor : '',
            width: 430 * (props.scale ? props.scale : 1),
            height: 270 * (props.scale ? props.scale : 1),
            padding: 15 * (props.scale ? props.scale : 1)
        } },
        React.createElement("div", { className: "component-atm-card-controls" },
            React.createElement("i", { onClick: () => props.onRotate ? props.onRotate() : {} },
                React.createElement(Rotate, { size: 34 * (props.scale ? props.scale : 1), color: props.dataColor ? props.dataColor : '' }))),
        React.createElement("div", { className: props.dark ? 'component-atm-card-magnet dark' : 'component-atm-card-magnet', style: { height: 60 * (props.scale ? props.scale : 1) } }),
        React.createElement("div", { className: "component-atm-card-bank-back" }, props.bankLogo),
        React.createElement("div", { className: "component-atm-card-cvv", style: {
                fontSize: 25 * (props.scale ? props.scale : 1),
                fontFamily: props.fontFamily ? props.fontFamily : '',
                color: props.dataColor ? props.dataColor : ''
            } },
            React.createElement("span", { style: { fontSize: 18 * (props.scale ? props.scale : 1) } }, "CVV CODE:"),
            !cvvEdit ?
                React.createElement("div", { onClick: () => {
                        var _a;
                        setCvvEdit(true);
                        (_a = cvvInput.current) === null || _a === void 0 ? void 0 : _a.focus();
                    } }, cvvMask(props.cvv)) :
                React.createElement(InputComponent, { type: "number", fontFamily: props.fontFamily ? props.fontFamily : '', fontSize: 25 * (props.scale ? props.scale : 1), onBlur: () => setCvvEdit(false), value: props.cvv, scale: props.scale, onKeyPress: handleKeyPress, onChange: (value) => props.onChange(value, 'cvv') })),
        React.createElement("div", { className: "component-atm-card-system-logo" }, props.system)));
};
export default BackCardData;
